src/library/scala/collection/concurrent/TrieMap.scala

```diff
@@ -1008,6 +1008,24 @@ final class TrieMap[K, V] private (r: AnyRef, rtupd: AtomicReferenceFieldUpdater
     r.cachedSize(this)
   }
 
+  private[collection] def filterInPlaceImpl(p: (K, V) => Boolean): this.type = {
+    val it = readOnlySnapshot().iterator
+    while (it.hasNext) {
+      val (k, v) = it.next()
+      if (p(k, v)) remove(k, v) // does unnecessary equality checking
```

Comment:
what's the unnecessary check? using `==` instead of `eq` when comparing the `v`?

Reply:
correct. unfortunately, the contract of `remove` seems to require `==`. at least given that we implement it for `JConcurrentMapWrapper` using a Java method that uses `equals`